### ISSUE TITLE
Next/script inside of _document.js won't load

### ISSUE THREAD

**Issue by timfuhrmann** [vercel/next.js#37225]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

```
    Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 21.4.0: Fri Mar 18 00:46:32 PDT 2022; root:xnu-8020.101.4~15/RELEASE_ARM64_T6000
    Binaries:
      Node: 17.7.2
      npm: 8.5.2
      Yarn: 1.22.18
      pnpm: N/A
    Relevant packages:
      next: 12.1.7-canary.16
      react: 17.0.2
      react-dom: 17.0.2

```

### What browser are you using? (if relevant)

Chrome 101.0.4951.64

### How are you deploying your application? (if relevant)

next dev & next start

### Describe the Bug

Usually I've integrated my `beforeInteractive` `next/script`'s outside of `_document.tsx`, but since the docs are saying that it's actually supposed to be integrated within `_document.tsx`, I moved my consent manager. Though it looks just like in the docs, the script won't load - it won't even appear inside of the documents head. Outside of `_document.tsx` the script works properly.

### Expected Behavior

The script should load.

### To Reproduce

```
import Script from "next/script";
import { Head, Html, Main, NextScript } from "next/document";

export default function Document() {
    return (
        <Html>
            <Head />
            <body>
                <Main />
                <NextScript />
                <Script
                    id="usercentrics-cmp"
                    src="https://app.usercentrics.eu/browser-ui/latest/loader.js"
                    data-settings-id="..."
                    strategy="beforeInteractive"
                />
            </body>
        </Html>
    );
}
```

Same result for every script I have tested.

**Comment by await-ovo** [comment]
It looks like the Script that uses the beforeInteractive strategy can only be placed in the` _document` `Head`: https://github.com/vercel/next.js/pull/37084

**Comment by timfuhrmann** [comment]
> It looks like the Script that uses the beforeInteractive strategy can only be placed in the` _document` `Head`: #37084

Unfortunately moving `<Script/>` to the document's `Head` didn't help either. Somehow it will only load within custom components, but not inside of `_document.tsx`.

**Comment by timfuhrmann** [comment]
> It looks like the Script that uses the beforeInteractive strategy can only be placed in the` _document` `Head`: #37084

Ahh yes, for canary it seems to be working if placed inside of `Head`. The stable version still fails to load no matter where `Script` is placed within `document.tsx`.  I guess this means that only the docs are wrong and it will be fixed?

**Comment by housseindjirdeh** [comment]
Yep recent changes that have landed to better support streaming restricted `beforeInteractive` usage in the `_document` `Head`. Docs are currently not reflecting this, but I'll be submitting a patch this week to remove the restriction (i.e. use Script with `beforeInteractive` anywhere in `_document.js`)
